src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala

```diff
@@ -52,6 +60,44 @@ object EMD extends DebugEnhancedLogging {
     }
   }
 
+  def addAgreementFields(emd: EasyMetadata)(implicit s: Settings): Unit = {
+    val agreementPath = new File(s.bagitDir, depositorInfoDir.resolve("agreements.xml").toString)
+    if (agreementPath.exists) {
+      val agreementsXml = XML.loadFile(agreementPath)
+      if (BooleanUtils.toBoolean((agreementsXml \\ "depositAgreementAccepted").text)) {
+        emd.getEmdRights.setAcceptedLicense(true)
+      }
+      addPrivacySensitiveRemark(emd, agreementsXml)
+    }
+    else {
+      logger.info("agreements.xml not found, not setting agreement data")
+    }
+  }
+
+  private def addPrivacySensitiveRemark(emd: EasyMetadata, agreementsXml: Elem): Unit = {
+    val signerId = (agreementsXml \\ "signerId").text
+    Option((agreementsXml \\ "containsPrivacySensitiveData").text) match {
+      case Some(boolText) =>
+        val privacySensitivePart = if (BooleanUtils.toBoolean(boolText)) "DOES"
+                                   else "DOES NOT"
+        emd.getEmdOther.getEasRemarks.add(new BasicRemark(s"Message for the Datamanager: according to the depositor $signerId this dataset $privacySensitivePart contain Privacy Sensitive data."))
+      case None =>
+        logger.info("The field containsPrivacySensitiveData could not be found in agreements.xml")
+        emd.getEmdOther.getEasRemarks.add(new BasicRemark(s"Message for the Datamanager: it could not be determined if this dataset does contain Privacy Sensitive data."))
+    }
+  }
+
+  private def addMessageForDataManager(emd: EasyMetadata)(implicit s: Settings): Unit = {
+    val msgForDataManager = new File(s.bagitDir, depositorInfoDir.resolve("message-from-depositor.txt").toString)
+    if (msgForDataManager.exists) {
+      val content = Source.fromFile(msgForDataManager).mkString
```

Comment:
IntelliJ tells me you didn't close this resource! Usually, when we use `better-files`, you could use `file.contentAsString`, but that's not the case right here. Hence you can use `new String(Files.readAllBytes(msgForDataManager), StandardCharsets.UTF_8)`.

So I refactored the method a bit:

```scala
private def addMessageForDataManager(emd: EasyMetadata)(implicit s: Settings): Unit = {
  val msgForDataManager = s.bagitDir.toPath.resolve(depositorInfoDir).resolve("message-from-depositor.txt")
  if (Files.exists(msgForDataManager)) {
    val content = new String(Files.readAllBytes(msgForDataManager), StandardCharsets.UTF_8)
    emd.getEmdOther.getEasRemarks.add(new BasicRemark(s"Message for the Datamanager: $content"))
  }
  else {
    logger.info("message-from-depositor.txt not found, not setting a remark")
  }
}
```